.github/workflows/docker-build-and-test.yml

```diff
@@ -0,0 +1,104 @@
+name: Docker Build and Test
+
+on:
+  workflow_dispatch:
+    inputs:
+      run_tests:
+        description: 'Run tests'
+        required: false
+  push:
+    branches: [ "improvement/docker-testing-workflow-branch" ]  # Isolated branch name for testing to ensure we don't affect main
+
+jobs:
+  build_and_test:
+    if: ${{ github.ref == 'refs/heads/improvement/docker-testing-workflow-branch' || github.event_name == 'workflow_dispatch' }}
```

Comment:
Why do we need this if, when you have already declared both events above?

Reply:
This was for additional safety while testing. Happy to remove once happy with additional functionality.

Reply:
Added